### ISSUE TITLE
AppLocker prevents package installation

### ISSUE THREAD

**Issue by michha** [microsoft/winget-cli#1208]
# Brief description of your issue

- our company has AppLocker enabled with an allow-list of directories that executables can be started
- when installing a package (like Git.Git) it results in the following error (manually translated into english)
> Installer-Hash successfully validated
> Package installation is going to start...
> Unexpected error during execution of command:
> 0x8007029c : An assert error has occured.
- the applocker event log states
> The execution from %OSDRIVE%\USERS\SOMEUSER\APPDATA\LOCAL\PACKAGES\MICROSOFT.DESKTOPAPPINSTALLER_8WEKYB3D8BBW              E\TEMPSTATE\WINGET\GIT.GIT.2.32.0.EXE was blocked.

# Expected behavior

We should be able to set the package directory where the packages are downloaded and executed.
A option within `winget settings` should be added to override the default location.

# Actual behavior
With AppLocker enabled, we can not use winget, because executables cannot be run inside the users profile path.

# Environment

```none
Windows Package Manager v1.0.11451
Windows: Windows.Desktop v10.0.19042.1052
Paket: Microsoft.DesktopAppInstaller v1.11.11451.0
```

**Comment by denelon** [comment]
We have a change coming in the next release to change the download location to %TEMP% https://github.com/microsoft/winget-cli/pull/1146.

**Comment by denelon** [comment]
This looks like it might be the same thing as #970.

**Comment by michha** [comment]
> We have a change coming in the next release to change the download location to %TEMP% https://github.com/microsoft/winget-cli/pull/1146.
> 
> 

This would not help us as the temp directory is the main directory for malicious scripts and programs and therefore also blocked by AppLocker.

Like I said, everyone with AppLocker needs a configuration option for the path from which the exe, msi and so on will be started.

**Comment by denelon** [comment]
Based on your comment, this does look like a duplicate of #970.

**Comment by denelon** [comment]
Duplicate of #970

**Comment by paalders** [comment]
@denelon I stubled across this issue as well. In my opinion this is not the same as #970. This issue is about the temp folder that is used by winget to download and run the installer. #970 is about the target directory of the software to be installed into. Both of these locations are blocked by AppLocker software, since all of them base in %LOCALAPPDATA%. The applications are installed in \Programs\ and the installer is run from \Temp\.

As a target install directory I can image people might want to install into the C:\Program Files\ folder, or something else. This location a user will not want to use for the temporary data stored by winget during installation, which this issue is about. I can image that you'd might to set this folder to C:\Temp\ or something different.

**Comment by denelon** [comment]
@paalders I'll discuss this with the team. If there aren't any major blocking reasons associated with a user specified location, I'll re-open the Issue and mark the appropriate comments "No longer applicable". If not, I'll share the reasons so we can look for alternative solutions. One of the benefits we might lose is the ability to use the "Disk Cleanup" behavior associated with "Temporary Files".

**Comment by denelon** [comment]
We still have some concerns regarding long path names, but it looks like this is a different ask than what I initially understood.

**Comment by michha** [comment]
@denelon Whats the status about this issue. We are still struggling with winget because we cannot set the temporary folder for the package content.

**Comment by denelon** [comment]
We will be tackling standalone .exe and .zip packages in 1.3. I'm highly expecting we will be looking at how we handle paths for those types of packages. It will likely make sense to bring this in with that effort. We will also need to include the information via `winget --info` as logs from some installers tend to get placed in the directory the installer runs from.

**Comment by zsmbrvr** [comment]
@denelon any update on this? We also use software restriction policy and %temp% is not and won't be in the allowed list.

**Comment by bpe71** [comment]
This didn't make it in v1.4?

**Comment by denelon** [comment]
Sadly, no. We did get .zip support included in 1.4, but there wasn't enough time to look at supporting alternate paths for where we download and run installers from.

**Comment by michha** [comment]
In the latest 1.5 preview I couldnt find any reference to this issue.
Will there be a fix for this issue in 1.5?

We really need an environment variable to give winget a hint for the temporary folder it should use.

**Comment by denelon** [comment]
@michha take a look at the [Roadmap](https://github.com/microsoft/winget-cli/discussions/2063) to see how we generally prioritize work. This Issue simply doesn't have enough "demand" yet to move up very high in the backlog. I've moved it into the v.next milestone rather than the backlog to see if there is more interest or need for this work.

**Comment by Trenly** [comment]
[Policy] Area-Settings

**Comment by denelon** [comment]
Status and a clarification on scope, since this comes up for AppLocker/WDAC environments.

Current behavior (verified on `master`): during a normal `winget install` / `winget upgrade`, the installer is downloaded to and executed from a hard-coded temporary location (`Runtime::PathName::Temp`) - it is not configurable. The `downloadDefaultDirectory` setting and `winget download --download-directory` only relocate the *download-only* output of the `winget download` command; they do not change where installers are staged and run during a normal install. So the scenario in this issue - installers being blocked because they run from a user-writable path - is still not addressed.

To be clear about intent: this is a legitimate enterprise enablement request, not a way to bypass AppLocker. Blocking execution from user-writable paths like `%LOCALAPPDATA%` and `%TEMP%` is a recommended hardening practice, and relocating WinGet's working directory does not weaken it - AppLocker rules are path/publisher/hash based, so a new location only helps if the administrator has already allow-listed it. The goal is to let WinGet operate *within* a hardened policy, not around it.

Because of that, the right shape for this is an administrator-controlled, machine-scoped working/staging directory - ideally surfaced through the existing Group Policy framework rather than as a per-user override - pointing at a directory the admin has allow-listed and ACL-protected against standard-user tampering. That keeps the control in the administrator's hands and avoids creating a user-writable-and-executable hole.

Keeping this open as an accepted enhancement with that framing.

_(Drafted with GitHub Copilot assistance.)_
